modal/_function_utils.py

```diff
@@ -186,10 +193,18 @@ def is_nullary(self):
         return all(param.default is not param.empty for param in self.signature.parameters.values())
 
 
+class LocalFunctionError(Exception):
+    pass
+
+
 def load_function_from_module(module, qual_name):
     # The function might be defined inside a class scope (e.g mymodule.MyClass.f)
     objs: list[Any] = [module]
     for path in qual_name.split("."):
+        if path == "<locals>":
```

Comment:
there is already some code that checks this (and raises `InvalidError` in the `FunctionInfo` constructor – maybe worth making it a utility function

Reply:
Actually, I'm confused how this exception could happen? The `load_function_from_module` is only called when `definition_type == api_pb2.Function.DEFINITION_TYPE_FILE` right? and if that's the case, the function shouldn't have been possible to create in the first place (because of the check I just mentioned)

Reply:
I thought so too at first, but it's also called for *only* extracting the lifecycle class of a serialized function, and then only working if the serialized function is also global... definitely feels like a code wart though, I agree.

I have a recollection of working on an alternative way of getting lifecycle classes for serialized functions at some point, but it will be a lot easier once we have class decorators or similar.